### PR TITLE
[unittest] BoundedDeque: Add single element test

### PR DESCRIPTION
**Pull request by tomchy**
I want to use single element queue, but it doesn't work. See unit test in this PR.

@daniel-k told me to do so :slightly_smiling_face:

### CODE REVIEW

**Comment by tomchy** [comment]
It seems that changing static assignment of tail fixes this issue (still should be checked):
```c
this->tail = (N == 1 ? 0 : 1);
```

This assignment can be found inside constructor and clear() method.

**Comment by salkinium** [comment]
Is this PR missing the commit with the fix?

**Comment by tomchy** [comment]
Yes. I've just pushed a simple fix. Please check :slightly_smiling_face:

File: src/xpcc/container/deque_impl.hpp [hunk 1]
```diff
@@ -36,8 +36,9 @@
 
 template<typename T, std::size_t N>
 xpcc::BoundedDeque<T, N>::BoundedDeque() : 
-	head(0), tail(1), size(0)
+	head(0), size(0)
```
Comment (salkinium) [inline src/xpcc/container/deque_impl.hpp]: I'd like this still in the constructor list, that will likely result in one less constructor call for static instantiations.
```diff
-	head(0), tail(1), size(0)
+	head(0), tail(N == 1 ? 0 : 1), size(0)
```

**Review by salkinium — CHANGES_REQUESTED** [review]
Apart from the minor change, this is 👍.

**Comment by tomchy** [comment]
Changes applied :slightly_smiling_face:

**Review by salkinium — APPROVED** [review]

**Comment by daniel-k** [comment]
👍

**Comment by salkinium** [comment]
I've squashed and rebased your commit for ~yak~ git history shaving purposes.

**Comment by salkinium** [comment]
Thanks a lot for the fix!

**Pull request merged**